src/main/AuthenticationRequest.ts

```diff
@@ -66,30 +105,40 @@ export default class AuthenticationRequest {
    * @param jwt
    * @param opts
    */
-  static async verifyJWT(jwt: string, opts: SIOP.VerifyAuthenticationRequestOpts): Promise<SIOP.VerifiedAuthenticationRequestWithJWT> {
+  static async verifyJWT(
+    jwt: string,
+    opts: VerifyAuthenticationRequestOpts,
+    linkedDomainValidationMode?: LinkedDomainValidationMode
+  ): Promise<VerifiedAuthenticationRequestWithJWT> {
     assertValidVerifyOpts(opts);
     if (!jwt) {
       throw new Error(SIOPErrors.NO_JWT);
     }
 
-    const { header, payload } = DIDJwt.parseJWT(jwt);
+    const { header, payload } = parseJWT(jwt);
     assertValidRequestJWT(header, payload);
 
     const options = {
-      audience: DIDJwt.getAudience(jwt),
+      audience: getAudience(jwt),
     };
 
     const verPayload = payload as AuthenticationRequestPayload;
     if (opts.nonce && verPayload.nonce !== opts.nonce) {
       throw new Error(`${SIOPErrors.BAD_NONCE} payload: ${payload.nonce}, supplied: ${opts.nonce}`);
-    } else if (verPayload.registration?.subject_syntax_types_supported && verPayload.registration.subject_syntax_types_supported.length == 0) {
-      throw new Error(`${SIOPErrors.VERIFY_BAD_PARAMS}`);
     }
 
-    const verifiedJWT = await DIDJwt.verifyDidJWT(jwt, DIDres.getResolver(opts.verification.resolveOpts), options);
+    AuthenticationRequest.assertValidRequestObject(verPayload);
+    AuthenticationRequest.assertValidRegistrationObject(
+      await AuthenticationRequest.getRegistrationObj(verPayload.registration_uri, verPayload.registration)
+    );
+
+    const verifiedJWT = await verifyDidJWT(jwt, getResolver(opts.verification.resolveOpts), options);
     if (!verifiedJWT || !verifiedJWT.payload) {
       throw Error(SIOPErrors.ERROR_VERIFYING_SIGNATURE);
     }
+    if (linkedDomainValidationMode && linkedDomainValidationMode != LinkedDomainValidationMode.NEVER) {
+      validateWithDid(verPayload.iss, linkedDomainValidationMode);
```

Comment:
Are we really sure the iss field is a DID here?

Reply:
yes, we're populating it with a did. of course if we were checking the `registration.issuer` it would be: `ResponseIss.SELF_ISSUED_V2`